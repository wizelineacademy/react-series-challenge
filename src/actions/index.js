export * from './favorite'
export * from './home'
export * from './searchbar'
export * from './detail'
