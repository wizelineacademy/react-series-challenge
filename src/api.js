export default {
    HOST: 'https://api.giphy.com',
    KEY: 'DJ3xIqeGfmTI3h1iNMK9uDeyLnAZWmxD',
    endpoints: {
        SEARCH: '/v1/gifs/search',
        TRENDING: '/v1/gifs/trending'
    }
}