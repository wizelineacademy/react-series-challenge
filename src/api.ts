import axios from 'axios';

const API_URL = "https://api.giphy.com/v1/gifs";
const API_KEY = process.env.REACT_APP_GIPHY_KEY;

const API = axios.create({ baseURL: API_URL, params: { api_key: API_KEY } });

export function getTrending() {
    return API.get('/trending');
}

export function search(query: string) {
    return API.get('/search', {
        params: {
            q: query,
            limit: 25,
            offset: 0,
            rating: 'G',
            lang: 'en'
        }
    });
}