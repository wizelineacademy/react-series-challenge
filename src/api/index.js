const API_ROOT = "http://api.giphy.com/v1/gifs";

const API_SEARCH = `${API_ROOT}/search`;
const API_TRENDING = `${API_ROOT}/trending`;

const buildQueryParams = params =>
  Object.keys(params).reduce(
    (string, key) => string.concat(key, "=", params[key], "&"),
    "?"
  );

const request = endpoint => mapping => {
  const config = Object.assign(mapping, {
    api_key: process.env.REACT_APP_API_KEY
  });
  const params = buildQueryParams(config);
  const query = endpoint.concat(params);
  return fetch(query).then(response => response.json());
};

const search = request(API_SEARCH);
const trending = request(API_TRENDING);

const searchGifs = ({ ...config }) => search(config);
const getTrendingGifs = ({ ...config }) => trending(config);
const getGifById = ({ id, ...config }) => request(`${API_ROOT}/${id}`)(config);

export { searchGifs, getTrendingGifs, getGifById };
