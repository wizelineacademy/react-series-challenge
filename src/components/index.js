import Card from './Card'
import Detail from './Detail'
import GifItem from './GifItem'
import Grid from './Grid'
import Header from './Header'
import Home from './Home'
import SearchBar from './SearchBar'

export {
  Card,
  Detail,
  GifItem,
  Grid,
  Header,
  Home,
  SearchBar
}
