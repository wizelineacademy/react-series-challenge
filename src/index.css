body, html, #root {
    background: black;
    font-family: sans-serif;
    height: auto;
    margin: 0;
    padding: 0;
    width: 100%;
}
