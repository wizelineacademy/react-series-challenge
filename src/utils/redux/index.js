import { combineReducers } from 'redux';

import pages from '../../pages/data/reducers';

export default combineReducers({
  pages,
});
