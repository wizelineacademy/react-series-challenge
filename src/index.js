import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly'

import './utils/base.css'

import rootReducer from './reducers'
import rootSaga from './sagas'
import Routes from './Routes'
import registerServiceWorker from './registerServiceWorker'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
  rootReducer,
  composeWithDevTools(),
  applyMiddleware(sagaMiddleware)
)
sagaMiddleware.run(rootSaga)


ReactDOM.render(
  <Provider store={store}>
    <Routes />
  </Provider>,
  document.getElementById('root')
)
registerServiceWorker()
