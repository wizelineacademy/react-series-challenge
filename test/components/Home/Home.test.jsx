import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { MemoryRouter } from 'react-router'
import { WrapperProvider } from '../../mocks/redux'

import { Home } from '../../../src/components'

const props = {
  giphies: [
    {
      id: '123',
      title: 'First title',
      images: {
        fixed_height: {
          url: 'someUrl.com'
        }
      }
    },
    {
      id: '1234',
      title: 'Second Title',
      images: {
        fixed_height: {
          url: 'someOtherUrl.com'
        }
      }
    }
  ],
  initialSetup: jest.fn(),
  keywordFilter: '',
  onSearchChange: jest.fn(),
  onSearchSubmit: jest.fn(),
  searchValue: ''
}

describe('Home', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Home {...props}/>
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('calls initialSetup function when rendered', () => {
    mount(
      <WrapperProvider>
        <MemoryRouter>
          <Home {...props} />
        </MemoryRouter>
      </WrapperProvider>
    )

    expect(props.initialSetup).toHaveBeenCalled()
  })
})
