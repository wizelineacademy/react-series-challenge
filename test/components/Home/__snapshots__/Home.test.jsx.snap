// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <Header>
    <SearchBar
      onChange={[Function]}
      onSubmit={[Function]}
      value=""
    />
  </Header>
  <Grid
    giphies={
      Array [
        Object {
          "id": "123",
          "images": Object {
            "fixed_height": Object {
              "url": "someUrl.com",
            },
          },
          "title": "First title",
        },
        Object {
          "id": "1234",
          "images": Object {
            "fixed_height": Object {
              "url": "someOtherUrl.com",
            },
          },
          "title": "Second Title",
        },
      ]
    }
  />
</div>
`;
